test/extensions/filters/http/cache/cache_filter_integration_test.cc

```diff
@@ -597,6 +598,54 @@ TEST_P(CacheIntegrationTest, ServeGetFollowedByHead200WithValidation) {
   }
 }
 
+ConfigHelper::HttpModifierFunction
+overrideConfig(const std::string& json_config,
+               const std::vector<std::string>& allowed_vary_headers) {
+  envoy::extensions::filters::http::cache::v3::CacheConfig cache_per_route;
+  TestUtility::loadFromJson(json_config, cache_per_route);
+
+  for (auto& v : allowed_vary_headers) {
+    const auto a = cache_per_route.mutable_allowed_vary_headers()->Add();
+    a->set_exact(v);
+  }
+
+  return
+      [cache_per_route](
+          envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+              cfg) {
+        auto* config = cfg.mutable_route_config()
+                           ->mutable_virtual_hosts()
+                           ->Mutable(0)
+                           ->mutable_typed_per_filter_config();
+
+        (*config)["envoy.filters.http.cache"].PackFrom(cache_per_route);
+      };
+}
+
+TEST_P(CacheIntegrationTest, RouteOverride) {
```

Comment:
How does this test verify that the override in vhost worked?

It might be easier to usefully test that route_config works in a unit test. I think it's config_test in file_system_buffer_filter that does some of that.

Reply:
added unittest